packages/core/src/components/Form.tsx

```diff
@@ -382,7 +395,8 @@ export default class Form<
     const _retrievedSchema = retrievedSchema ?? schemaUtils.retrieveSchema(schema, formData);
 
     const getCurrentErrors = (): ValidationData<T> => {
-      if (props.noValidate) {
+      // If the `props.noValidate` option is set or the schema has changed, we reset the error state.
+      if (props.noValidate || isSchemaChanged) {
```

Comment:
Should we validate against the new schema instead of clearing the errors if `mustValidate` is true?
```suggestion
      if (props.noValidate || (isSchemaChanged && !mustValidate)) {
```